auction-app-backend/src/main/java/com/atlantbh/auctionappbackend/service/ProductService.java

```diff
@@ -94,14 +99,66 @@ public PaginatedResponse<Product> getAll(
                 categoryIds, categoriesAvailable, minPrice, maxPrice, search, PageRequest.of(page, size, sort)
         );
 
-        return new PaginatedResponse<>(
-                pageProducts.getContent(),
-                pageProducts.getNumber(),
-                pageProducts.getTotalElements(),
-                pageProducts.getTotalPages()
+        String searchSuggestion = null;
+        if (search != null && pageProducts.getTotalElements() == 0) {
+            searchSuggestion = getSearchSuggestion(search);
+        }
+
+        return new ProductsResponse(
+                new PaginatedResponse<>(
+                        pageProducts.getContent(),
+                        pageProducts.getNumber(),
+                        pageProducts.getTotalElements(),
+                        pageProducts.getTotalPages()
+                ),
+                searchSuggestion
         );
     }
 
+    private String getSearchSuggestion(String searchedValue) {
```

Comment:
To make it easier to understand and maintain, I suggest adding Javadoc for this method to explain how we get search suggestion.